src/main/java/ladysnake/requiem/common/block/RunicObsidianBlockEntity.java

```diff
@@ -0,0 +1,307 @@
+            this.refresh();
+
+            Vec3d obeliskCenter = new Vec3d(
+                MathHelper.lerp(0.5, this.pos.getX(), this.pos.getX() + obeliskWidth - 1),
+                this.pos.getY() - 2,
+                MathHelper.lerp(0.5, this.pos.getZ(), this.pos.getZ() + obeliskWidth - 1)
+            );
+
+            if (!this.levels.isEmpty() && this.findPowerSource((ServerWorld) this.world, obeliskCenter, this.obeliskWidth * 5)) {
+                this.applyPlayerEffects();
+            }
+        }
+    }
+
+    private void applyPlayerEffects() {
+        Preconditions.checkState(this.world != null);
+        Preconditions.checkState(this.obeliskWidth > 0);
+        Preconditions.checkState(this.obeliskHeight > 0);
+
+        double range = this.obeliskWidth * 10 + 10;
+        Box box = (new Box(this.pos, this.pos.add(obeliskWidth-1, obeliskHeight-1, obeliskWidth-1))).expand(range);
+
+        int effectDuration = (9 + this.obeliskWidth * 2) * 20;
+        List<PlayerEntity> players = this.world.getNonSpectatingEntities(PlayerEntity.class, box);
+
+        for (PlayerEntity player : players) {
+            if (RemnantComponent.get(player).getRemnantType().isDemon()) {
+                for (Object2IntMap.Entry<StatusEffect> effect : this.levels.object2IntEntrySet()) {
+                    player.addStatusEffect(new StatusEffectInstance(effect.getKey(), effectDuration, effect.getIntValue() - 1, true, true));
+                }
+            }
+        }
+    }
+
+    private boolean findPowerSource(ServerWorld world, Vec3d center, double range) {
+        BlockPos.Mutable checked = new BlockPos.Mutable();
+
+        for (int attempt = 0; attempt < RunicObsidianBlockEntity.POWER_ATTEMPTS; attempt++) {
+            // https://stackoverflow.com/questions/5837572/generate-a-random-point-within-a-circle-uniformly/50746409#50746409
+            double r = range * Math.sqrt(world.random.nextDouble());
+            double theta = world.random.nextDouble() * 2 * Math.PI;
+            double x = center.x + r * Math.cos(theta);
+            double z = center.z + r * Math.sin(theta);
+            checked.set(Math.round(x), center.y, Math.round(z));
+            BlockState state = world.getBlockState(checked);
+
+            while (!state.isSolidBlock(world, checked)) {
+                checked.move(Direction.DOWN);
+                state = world.getBlockState(checked);
+            }
+
+            if (state.isIn(BlockTags.SOUL_SPEED_BLOCKS)) {
+                Vec3d particleSrc = new Vec3d(checked.getX() + 0.5, checked.getY() + 0.9, checked.getZ() + 0.5);
+                Vec3d toObelisk = center.subtract(particleSrc).normalize();
+                world.spawnParticles(ParticleTypes.SOUL, particleSrc.x, particleSrc.y, particleSrc.z, 0, toObelisk.x, 1, toObelisk.z, 0.1);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private RunicObsidianBlockEntity getObeliskOrigin() {
```

Comment:
Could this cause issues with sections of obelisks?

Reply:
Ah yes, this may be an issue. I will have to check how beacons validate their structure.

Reply:
okay, but if an issue pops up, I get to laugh at you :P